### PR TITLE
fix(op):skip db consistency check for op mainnet

### LINKED ISSUE #9725: op-reth, optimism, unable to run the node after completing bedrock import
### Describe the bug

```
/target/maxperf/op-reth node --chain=optimism --datadir=/data
2024-07-23T06:42:35.476348Z  INFO Initialized tracing, debug log directory: .cache/reth/logs/optimism
2024-07-23T06:42:35.482860Z  INFO Starting reth version="1.0.3 (ed16643f)"
2024-07-23T06:42:35.483550Z  INFO Opening database path="/data/db"
2024-07-23T06:42:35.541843Z  INFO Configuration loaded path="/data/reth.toml"
2024-07-23T06:42:35.559866Z  INFO Verifying storage consistency.
2024-07-23T06:42:35.589673Z  INFO Setting unwind target. checkpoint_block_number=105235063 unwind_target=0 segment=Receipts
thread 'main' panicked at reth/crates/node/builder/src/launch/common.rs:384:13:
assertion `left != right` failed: A static file <> database inconsistency was found that would trigger an unwind to block 0
  left: Unwind(0)
 right: Unwind(0)
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
When trying to start an optimism node after importing the bedrock data the following error is returned.

### Steps to reproduce

1) export blocks from 0 to 105235063 using op-geth
2) export receipts from 1 to 105235063 using op-geth(bumped to 1 to work around the other op-reth issue I have openned)
3) export state
4) import blocks, receipts and state using import-op import-receipts-op and init-state.
5) start node error from above is returned

The same issue can be recreated without importing receipts either

### Node logs

```text
logs also include the tail end of importing state

2024-07-23T06:12:40.090033Z  INFO reth::cli: parsed_new_accounts=3137508
2024-07-23T06:12:54.377297Z  INFO reth::cli: parsed_new_accounts=3422736
2024-07-23T06:13:18.411355Z  INFO reth::cli: parsed_new_accounts=3707964
2024-07-23T06:13:34.493039Z  INFO reth::cli: parsed_new_accounts=3993192
2024-07-23T06:13:36.502559Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=285229
2024-07-23T06:14:07.099225Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=570457
2024-07-23T06:14:35.788961Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=855685
2024-07-23T06:15:21.952165Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=1140913
2024-07-23T06:16:25.976869Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=1426141
2024-07-23T06:17:04.043846Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=1711369
2024-07-23T06:18:03.764680Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=1996597
2024-07-23T06:19:23.442719Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=2281825
2024-07-23T06:20:04.206371Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=2567053
2024-07-23T06:21:08.899953Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=2852281
2024-07-23T06:21:36.199304Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=3137509
2024-07-23T06:22:19.333791Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=3422737
2024-07-23T06:22:46.421493Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=3707965
2024-07-23T06:23:27.121249Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=3993193
2024-07-23T06:24:41.082322Z  INFO reth::cli: Writing accounts to db total_inserted_accounts=4007565
2024-07-23T06:33:48.956872Z  INFO reth::cli: Computed state root matches state root in state dump computed_state_root=0x920314c198da844a041d63bf6cbe8b59583165fd2229d1b3f599da812fd424cb
2024-07-23T06:33:49.052775Z DEBUG storage::db::mdbx: Commit total_duration=95.843943ms commit_latency=Some(CommitLatency(MDBX_commit_latency { preparation: 1, gc_wallclock: 4038, audit: 0, write: 0, sync: 2221, ending: 8, whole: 6271, gc_cputime: 305, gc_prof: MDBX_commit_latency__bindgen_ty_1 { wloops: 0, coalescences: 0, wipes: 0, flushes: 0, kicks: 0, work_counter: 0, work_rtime_monotonic: 0, work_xtime_cpu: 0, work_rsteps: 0, work_xpages: 0, work_majflt: 0, self_counter: 0, self_rtime_monotonic: 0, self_xtime_cpu: 0, self_rsteps: 0, self_xpages: 0, self_majflt: 0 } })) is_read_only=false
2024-07-23T06:33:50.204668Z  INFO reth::cli: Genesis block written hash=0xdbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3
2024-07-23T06:36:42.374948Z  INFO reth::cli: Initialized tracing, debug log directory: /home/argakiig/.cache/reth/logs/optimism
2024-07-23T06:36:42.385907Z  INFO reth::cli: Starting reth version="1.0.3 (ed16643f)"
2024-07-23T06:36:42.388303Z  INFO reth::cli: Opening database path="/data/db"
2024-07-23T06:36:42.433088Z DEBUG reth_node_builder::launch::common: Raised file descriptor limit from=1048576 to=1048576
2024-07-23T06:36:42.455981Z  INFO reth::cli: Configuration loaded path="/data/reth.toml"
2024-07-23T06:36:42.489462Z  INFO reth::cli: Verifying storage consistency.
2024-07-23T06:36:42.521414Z  INFO reth::providers::static_file: Setting unwind target. checkpoint_block_number=105235063 unwind_target=0 segment=Receipts
```
```


### Platform(s)

Linux (x86)

### What version/commit are you on?

ed16643f have also tried v0.2.0-beta8 through 1.0.3 for builds without the bug blocking blocks importing


### What database version are you on?

2

### Which chain / network are you on?

optimism

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

cargo build --profile=maxperf --features=jemalloc,asm-keccak,optimism --locked --bin op-reth

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by emhane**
Closes https://github.com/paradigmxyz/reth/issues/9725

OVM chain is knowingly inconsistent, due to duplicate txns. Undefined behaviour for RPC queries for transactions below bedrock is allowed in op-reth, as well as op-erigon.

### CODE REVIEW

File: crates/storage/provider/src/providers/static_file/manager.rs [hunk 1]
```diff
@@ -532,6 +532,17 @@ impl StaticFileProvider {
         provider: &DatabaseProvider<TX>,
         has_receipt_pruning: bool,
     ) -> ProviderResult<Option<PipelineTarget>> {
+        // OVM chain contains duplicate transactions, so is inconsistent by default since reth db
+        // not designed for duplicate transactions (see <https://github.com/paradigmxyz/reth/blob/v1.0.3/crates/optimism/primitives/src/bedrock_import.rs>). Undefined behaviour for queries
+        // to OVM chain is also in op-erigon.
+        if provider.chain_spec().is_optimism_mainnet() {
+            info!(target: "reth::cli",
+                "Skipping storage verification for OP mainnet, expected inconsistency in OVM chain"
+            );
```
Comment (argakiig) [inline crates/storage/provider/src/providers/static_file/manager.rs:541]: it logs the expected inconsistence, but I think you meant to return Ok(None) here as well so check_consistency doesnt continue
```suggestion
            );
            return Ok(None);
```

Comment (mattsse) [reply crates/storage/provider/src/providers/static_file/manager.rs:541]: that makes sense

**Review by mattsse — COMMENTED** [review]
that makes sense

another appropriate location for this would be in the launch setup:

https://github.com/paradigmxyz/reth/blob/5a9de035bf779606247b3e67ae1907011e2339b7/crates/node/builder/src/launch/common.rs#L376-L381

which would perhaps be better?

**Comment by emhane** [comment]
> that makes sense
> 
> another appropriate location for this would be in the launch setup:
> 
> https://github.com/paradigmxyz/reth/blob/5a9de035bf779606247b3e67ae1907011e2339b7/crates/node/builder/src/launch/common.rs#L376-L381
> 
> which would perhaps be better?

the op mainnet db will be inconsistent, no matter which command runs it. no need for redundant code by having it in launch command and `EnvironmentArgs`.

**Review by mattsse — APPROVED** [review]
gotcha

**Pull request merged**